### PR TITLE
Document .NET Aspire support

### PR DESCRIPTION
**Pull request by mikeminutillo**
- [x] Sample

_replaces https://github.com/Particular/docs.particular.net/tree/aspire-sample_
_replaces https://github.com/Particular/docs.particular.net/pull/6839_

### CODE REVIEW

File: .github/workflows/validate-pull-requests.yml [hunk 1]
```diff
@@ -53,6 +53,12 @@ jobs:
         uses: actions/setup-dotnet@v4.0.1
         with:
           dotnet-version: 8.0.x
+      - name: Install .NET Aspire workload
+        run: |
+          echo "Updating advertising manifests"
+          dotnet workload update --advertising-manifests-only
+          echo "Installing the Apsire workload"
```
Comment (johnsimons) [inline .github/workflows/validate-pull-requests.yml:60]:
```suggestion
          echo "Installing the Aspire workload"
```

File: .github/workflows/build-samples-and-snippets.yml [hunk 2]
```diff
@@ -17,6 +17,12 @@ jobs:
         uses: actions/setup-dotnet@v4.0.1
         with:
           dotnet-version: 8.0.x
+      - name: Install .NET Aspire workload
+        run: |
+          echo "Updating advertising manifests"
+          dotnet workload update --advertising-manifests-only
+          echo "Installing the Apsire workload"
```
Comment (johnsimons) [inline .github/workflows/build-samples-and-snippets.yml:24]:
```suggestion
          echo "Installing the Aspire workload"
```

File: samples/hosting/aspire/Core_9/Sales/PlaceOrderHandler.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        public Task Handle(PlaceOrder message, IMessageHandlerContext context)
+        {
+            log.Info($"Received PlaceOrder, OrderId = {message.OrderId}");
+
+            // This is normally where some business logic would occur
+
+            // Uncomment to test throwing a systemic exception
+            //throw new Exception("BOOM");
+
+            //Uncomment to test throwing a transient exception
```
Comment (johnsimons) [inline samples/hosting/aspire/Core_9/Sales/PlaceOrderHandler.cs:27]:
```suggestion
```

File: samples/hosting/aspire/sample.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+---
+title: Hosting endpoints in .NET Aspire
+summary: Hosting multiple NServiceBus endpoints in an .NET Aspire application host
+component: Core
+reviewed: 2024-09-20
+---
+
+[.NET Aspire](https://learn.microsoft.com/en-us/dotnet/aspire/) is a stack for developing distributed applications provided by Microsoft.
+
+This sample shows a .NET Aspire AppHost project which orchestrates multiple NServiceBus endpoints, along with docker images providing a RabbitMQ transport and PostgreSQL persistence.
```
Comment (johnsimons) [inline samples/hosting/aspire/sample.md:10]:
```suggestion
This sample shows a .NET Aspire AppHost project that orchestrates multiple NServiceBus endpoints. It also launches the following infrastructure, a RabbitMQ server for the configured transport and a PostgreSQL server for persistence.
```

File: samples/hosting/aspire/Core_9/Sales/PlaceOrderHandler.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            if (random.Next(0, 5) == 0)
+            {
+                throw new Exception("Oops");
+            }
+
+            var publishOptions = new PublishOptions();
+            publishOptions.ContinueExistingTraceOnReceive();
```
Comment (mikeminutillo) [inline samples/hosting/aspire/Core_9/Sales/PlaceOrderHandler.cs:34]:
```suggestion
            // https://docs.particular.net/nservicebus/operations/opentelemetry#traces-emitted-span-structure-publish-operations
            publishOptions.ContinueExistingTraceOnReceive();
```

File: samples/hosting/aspire/Core_9/Sales/PlaceOrderHandler.cs (hunk 3 above)
Comment (mikeminutillo) [inline samples/hosting/aspire/Core_9/Sales/PlaceOrderHandler.cs:27]:
```suggestion
```

File: samples/hosting/aspire/sample.md (hunk 4 above)
Comment (mikeminutillo) [inline samples/hosting/aspire/sample.md:10]:
```suggestion
This sample shows a .NET Aspire AppHost project that orchestrates multiple NServiceBus endpoints, wiring up the required infrastructure pieces including a RabbitMQ broker and PostgreSQL database.
```

File: samples/hosting/aspire/sample.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+### AspireDemo.AppHost
+
+The [.NET Aspire orchestration project](https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/app-host-overview) defines multiple resources and the relationships between them:
+
+- A RabbitMQ instance named `transport`
+- A PostgreSQL server named `database`
+  - A database named `shipping-db`
+  - An instance of [pgweb](https://sosedoff.github.io/pgweb/) to access the database
+- 4 projects, each of which is an NServiceBus endpoint. All of these projects reference the `transport` resource.
```
Comment (mikeminutillo) [inline samples/hosting/aspire/sample.md:31]:
```suggestion
- Four projects, each of which is an NServiceBus endpoint. All of these projects reference the `transport` resource.
```

File: shape-the-future/aspire.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+---
+title: NServiceBus and Aspire
+summary: How to use NServiceBus with Aspire
+reviewed: 2024-09-16
+---
+
+[.NET Aspire](https://learn.microsoft.com/en-us/dotnet/aspire/get-started/aspire-overview) is an opinionated, cloud ready stack for building observable, production ready, distributed applications.​
```
Comment (mikeminutillo) [inline shape-the-future/aspire.md:7]:
```suggestion
[.NET Aspire](https://learn.microsoft.com/en-us/dotnet/aspire/get-started/aspire-overview) is an opinionated, cloud-ready stack for building observable, production ready, distributed applications.​
```

File: shape-the-future/aspire.md [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+
+NServiceBus endpoints can run under an Aspire AppHost orchestrator, we have a sample that shows you how it all works.
```
Comment (mikeminutillo) [inline shape-the-future/aspire.md:9]: Needs a link

Comment (mikeminutillo) [inline shape-the-future/aspire.md:9]:
```suggestion
NServiceBus endpoints can run under an Aspire AppHost orchestrator, we have [a sample that shows you how it all works](/samples/hosting/aspire/).
```

File: shape-the-future/aspire.md [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
+
+But I am sure we can do much more, and this is where you come in and tell us how would you like NServiceBus to be even more integrated with the .NET Aspire orchestrator framework.
+So if you have a spare moment to help us shape how NServiceBus should support Aspire, drop us a comment in [this](https://github.com/Particular/NServiceBus/issues/xxx) GitHub issue.
```
Comment (mikeminutillo) [inline shape-the-future/aspire.md:12]:
```suggestion
If you are using NServiceBus with .NET Aspire today, [tell us what's missing](https://github.com/Particular/NServiceBus/issues/6941).
```

File: shape-the-future/aspire.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+title: NServiceBus and Aspire
+summary: How to use NServiceBus with Aspire
+reviewed: 2024-09-16
+---
+
+[.NET Aspire](https://learn.microsoft.com/en-us/dotnet/aspire/get-started/aspire-overview) is an opinionated, cloud-ready stack for building observable, production ready, distributed applications.​
+
+NServiceBus endpoints can run under an Aspire AppHost orchestrator, we have [a sample that shows you how it all works](/samples/hosting/aspire/).
+
+If you are using NServiceBus with .NET Aspire today, [tell us what's missing](https://github.com/Particular/NServiceBus/issues/6941).
```
Comment (johnsimons) [inline shape-the-future/aspire.md:11]: We should update the issue to include the leading questions:
- Do you use https://learn.microsoft.com/en-us/dotnet/aspire/deployment/manifest-format
- Are you currently using Aspire, if so is it just for development or also for production deployments
- What other Aspire integrations are you using
- Do you use health checks - https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/health-checks
- Do you use telemetry

File: menu/menu.yaml [hunk 11]
```diff
@@ -1662,4 +1662,6 @@
   - Title: Introduction
     Url: shape-the-future
   - Title: NServiceBus and CloudEvents
-    Url: shape-the-future/cloudevents
\ No newline at end of file
+    Url: shape-the-future/cloudevents
+  - Title: NServiceBus and Aspire
```
Comment (johnsimons) [inline menu/menu.yaml:1666]:
```suggestion
  - Title: NServiceBus and .NET Aspire
```

File: shape-the-future/aspire.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
+---
+title: NServiceBus and Aspire
+summary: How to use NServiceBus with Aspire
```
Comment (johnsimons) [inline shape-the-future/aspire.md:3]:
```suggestion
title: NServiceBus and .NET Aspire
summary: How to use NServiceBus with .NET Aspire
```

**Review by johnsimons — APPROVED** [review]

**Pull request merged**